src/shared/compounders/AMMCompounderBase.sol

```diff
@@ -382,6 +387,15 @@ abstract contract AMMCompounderBase is ReentrancyGuard, ERC4626 {
 
     /********************************** Restricted Functions **********************************/
 
+    /// @dev Updates the feelessRedeemerWhitelist
+    /// @param _address - The address to update
+    /// @param _whitelist - The new whitelist status
+    function updateFeelessRedeemerWhitelist(address _address, bool _whitelist) external {
+        if (msg.sender != settings.owner) revert Unauthorized();
+
+        feelessRedeemerWhitelist[_address] = _whitelist;
```

Comment:
How does the best practice here look like with regards to adding and removing state? Since the contracts are immutable this implementation produces effective state bloat. The question might be if this state bloat is negligible over time. From an engineering point of view it feels wrong to only ever add state and never clean it up in case it does not serve a purpose anymore. What are the reasons not to effectively remove non-whitelisted addresses?

Reply:
The tradeoff here is gas cost vs "bloat":
- Using a dynamic array + cleaning it will cost an increasing amount of gas. Also it will require a separate `Redeem` function in order to not burden non-whitelisted users with the extra gas costs.
- The "bloating" is negligible, as mentioned here https://ethereum.stackexchange.com/questions/8650/is-there-a-maximum-number-of-entries-for-a-mapping.

Reply:
I get it. I hate it but that is probably just the good, the bad and the ugly of smart contract engineering today. Thanks for clarifying this.